kpi/deployment_backends/openrosa_backend.py

```diff
@@ -731,7 +733,13 @@ def get_validation_status(
 
     @property
     def mongo_userform_id(self):
-        return '{}_{}'.format(self.asset.owner.username, self.xform_id_string)
+        try:
+            return (
+                self.xform.mongo_uuid
+                or f'{self.asset.owner.username}_{self.xform_id_string}'
+            )
+        except KeyError:
```

Comment:
`KeyError`? when can this error happen?

Reply:
Thanks for the review Olivier. The `KeyError` can occur when the `formid` key is missing from the `backend_response` dictionary. This is typically the case in certain test scenarios, such as `ConflictingVersionsMockDataExports`, where mock data doesn't include the `formid` key.

When `self.xform` is accessed, it attempts to retrieve the `formid` from `backend_response`, which raises a `KeyError` if it's not present.

Reply:
As our per discussion, we don't want the `KeyError` to be silently ignored. 
It should never happen but if it does, retrieving `xform_id_string` should be also an issue. 

Fixture has been fixed to match production flow